packages/common/src/dotlottie-player.ts

```diff
@@ -294,15 +306,169 @@ export class DotLottiePlayer {
+            validatedOptions.direction = value as AnimationDirection;
+          }
+          break;
+
+        case 'loop':
+          if (typeof value === 'boolean') {
+            validatedOptions.loop = value;
+          }
+          break;
+
+        case 'playMode':
+          if (typeof value === 'string' && ['normal', 'bounce'].includes(value)) {
+            validatedOptions.playMode = value as PlayMode;
+          }
+          break;
+
+        case 'speed':
+          if (typeof value === 'number') {
+            validatedOptions.speed = value;
+          }
+          break;
+
+        case 'themeColor':
+          if (typeof value === 'string') {
+            validatedOptions.themeColor = value;
+          }
+          break;
+
+        case 'hover':
+          if (typeof value === 'boolean') {
+            validatedOptions.hover = value;
+          }
+          break;
+
+        case 'intermission':
+          if (typeof value === 'number') {
+            validatedOptions.intermission = value;
+          }
+          break;
+
+        default:
+          break;
+      }
+    }
+
+    return validatedOptions;
+  }
+
+  public play(activeAnimation?: string | number, options?: PlaybackOptions): void {
     if (!this._lottie) return;
 
-    if (this._lottie.playDirection === -1 && this._lottie.currentFrame === 0) {
-      this._lottie.goToAndPlay(this._lottie.totalFrames, true);
-    } else {
-      this._lottie.play();
+    if (!activeAnimation) {
+      if (this._lottie.playDirection === -1 && this._lottie.currentFrame === 0) {
+        this._lottie.goToAndPlay(this._lottie.totalFrames, true);
+      } else {
+        this._lottie.play();
+      }
+      this.setCurrentState(PlayerState.Playing);
+
+      return;
     }
-    this.setCurrentState(PlayerState.Playing);
+
+    if (typeof activeAnimation === 'number') {
+      const anim = this._manifest?.animations[activeAnimation];
+
+      if (!anim) return;
+      this.render(anim);
+    }
+
+    if (typeof activeAnimation === 'string') {
+      const anim = this._manifest?.animations.find((animation) => animation.id === activeAnimation);
+
+      if (!anim) return;
+      this.render({
+        ...anim,
+        ...this._validatePlaybackOptions(options),
+      });
+    }
+  }
+
+  public get activeAnimationId(): string | undefined {
+    return this._activeAnimationId;
+  }
+
+  public reset(): void {
+    const activeId = this._manifest?.activeAnimationId;
+
+    if (!activeId) {
+      const anim = this._manifest?.animations[0];
+
+      if (!anim || !anim.id) return;
+
+      this.render(anim);
+    }
+
+    const anim = this._manifest?.animations.find((animation) => animation.id === activeId);
+
+    if (!anim) return;
+
+    this.render(anim);
+  }
+
+  public prev(options?: PlaybackOptions): void {
+    if (!this._manifest || !this._manifest.animations.length) {
+      throw new Error('[dotlottie]: manifest not found.');
```

Comment:
maybe we can use util methods for easier Error management 

```

export const createError = (message: string): Error => {
  const error = new Error(`[dotLottie] ${message}`);

  return error;
};

export const logError = (message: string): void => {
  const error = `[dotLottie] ${message}`;

  console.error(error);
};

export const logWarning = (message: string): void => {
  const warning = `[dotLottie] ${message}`;

  console.warn(warning);
};

```

```
throw createError('...');
```

Reply:
✅ Done